analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.Microsoft.Extensions.Logging.cs

```diff
@@ -0,0 +1,405 @@
+        }
+    }
+
+    private void Log()
+    {
+        logger.LogCritical("Message!"); // Compliant - we do not check this
+    }
+
+    private void LogFromMultipleCatchBlocks()
+    {
+        try { }
+        catch (DivideByZeroException)
+        {
+            LoggerExtensions.LogCritical(logger, "Message!"); // Noncompliant
+        }
+        catch (AggregateException)
+        {
+            LoggerExtensions.LogCritical(logger, "Message!"); // Noncompliant
+        }
+        catch (ApplicationException e)
+        {
+            LoggerExtensions.LogCritical(logger, e, "Message!");
+        }
+    }
+
+    private void LogFromNestedCatchBlocks(Exception wrongException)
+    {
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning("Message!");                  // Noncompliant
+            logger.LogWarning(wrongException, "Message!");  // Noncompliant - wrong exception
+            try { }
+            catch (DivideByZeroException)
+            {
+                logger.LogCritical("Message!");             // Noncompliant
+            }
+            catch (AggregateException e1)
+            {
+                logger.LogCritical(e, "Message!");          // Noncompliant - wrong exception
+            }
+            catch (Exception e2)
+            {
+                logger.LogCritical(e2, "Message!");
+            }
+        }
+    }
+
+    private void ReAssignment()
+    {
+        try { }
+        catch (Exception e)
+        {
+            var other = e;
+            logger.LogWarning(other, "Message!");     // Noncompliant
+        }
+    }
+
+    private void Filtering()
+    {
+        try { }
+        catch (Exception e) when (e is InvalidCastException)
+        {
+            logger.LogWarning(e, "Message!");
+        }
+        catch (Exception e) when (e is DivideByZeroException divideByZeroException)
+        {
+            logger.LogWarning(divideByZeroException, "Message!");   // Noncompliant - FP the exception has other name
+        }
+        catch (Exception e) when (e is InvalidOperationException invalidOperationException || e is InvalidTimeZoneException invalidTimeZoneException)
+        {
+            logger.LogWarning(e, "Message!");                       // Compliant - the exception is logged (even if it has other names)
+        }
+    }
+
+    public void LogFromCatchBlockWithNoException()
+    {
+        try { }
+        catch
+        {
+            logger.LogCritical("Message!");           // Noncompliant
+        }
+    }
+
+    public void LogFromIfStatement()
+    {
+        try { }
+        catch (DivideByZeroException e)
+        {
+            if (true)
+            {
+                logger.LogCritical("Message!");       // Noncompliant
+            }
+        }
+        catch (Exception e)
+        {
+            if (true)
+            {
+                logger.LogCritical(e, "Message!");
+            }
+        }
+    }
+
+    public void LogFromCustomLogger()
```

Comment:
Also, add a test case where a custom logger implements the ILogger interface. Or just use [NullLogger](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.abstractions.nulllogger).